aws_lambda_powertools/shared/headers_serializer.py

```diff
@@ -0,0 +1,101 @@
+import warnings
+from collections import defaultdict
+from typing import Any, Dict, List
+
+
+class BaseHeadersSerializer:
+    """
+    Helper class to correctly serialize headers and cookies on the response payload.
```

Comment:
```suggestion
    Helper class to correctly serialize headers and cookies for Amazon API Gateway and ALB response payload.
```

Because this is in shared, we wouldn't know where this belongs unless you're familiar with the code base. This makes me think - Shouldn't this be closer to Event Handler in its own file as opposed to `shared`?

I did a poor job in documenting `shared` at the project level as it was meant to be used by more than one utility. When we get to modularization, we could turn this into a `commons` package.